src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtlDocumentTransformer.cs

```diff
@@ -233,38 +291,69 @@ public override void Transform(RavenEtlItem item, EtlStatsScope stats)
                         {
                             AddSingleCounterGroup(item.DocumentId, item.CounterGroupDocument);
                         }
-
+                        break;
+                    case EtlItemType.TimeSeries:
+                        if (_script.HasTransformation)
+                        {
+                            //TODO Is it make sense to run the main script (dry run) to know if the document is actual loaded to same collection. The second option is to check if the document can loaded to same collection according to load signature
```

Comment:
`_script.IsLoadedToDefaultCollection` just check if there is a possibility to load the doc to the same collection (check for the existence of loadTo<collection> in the script and compare `collection` to item collection).
But if the document is not actually loaded because of a condition.
Like this for example:
```
if(this.Name === 'SomeName')
    return;
var person = loadToUsers(this);
```
We will keep trying to send time-series any time they change (I think we do that in `counters` as well).
This can be a big overhead if most of the documents are not loaded (or not loaded to the same collection).
I added `_currentRun.IsDocumentLoadedToSameCollection(item.DocumentId)` in `EtlItemType.Document` case but I need the script to run to use this function.
I don't want to actually run the script so the doc will not reload over and over for any time-series change but maybe a dry run? I tried to play with it a little bit but I think this requires a big change in the code (but not sure) so I left the comment.

Reply:
This sounds like non trivial change so its a bit risky. It's shouldn't be part of adding TS to ETL for sure. Please create a separate ticket for that.